dev-tools/releng/src/main.rs

```diff
@@ -307,14 +342,19 @@ async fn main() -> Result<()> {
             .arg(&args.helios_dir)
             .ensure_success(&logger)
             .await?;
+        if let Some(helios) = &pins.helios {
+            info!(
+                logger,
+                "checking out {} to {}", args.helios_dir, helios.commit,
+            );
+            Command::new(&args.git_bin)
+                .arg("-C")
+                .arg(&args.helios_dir)
+                .args(["checkout", &helios.commit])
+                .ensure_success(&logger)
+                .await?;
```

Comment:
In the **rel/v\*** branches, I've been using `git reset --hard $commit`; e.g., see https://github.com/oxidecomputer/omicron/commit/020fde11e74b09202d6c53b491768bc4912903af

Is `git checkout` always equivalent to that?

Reply:
`git checkout` will fail if there's changes in the working tree, whereas `git reset --hard` will simply wipe them. In this branch we have freshly cloned the repo because the directory was nonexistent or empty, so in this particular case it should always be equivalent.

I'm hesitant to use `git reset --hard` even on a fresh clone out of paranoia, for the sake of running this tool outside of CI.

Reply:
Alright!